### PR TITLE
os/arch/arm/src/amebad: Add all RTK reboot reason

### PR DESCRIPTION
**Pull request by ziyik**
add all RTK reboot reason

### CODE REVIEW

File: os/arch/arm/src/amebad/amebad_reboot_reason.c [hunk 1]
```diff
@@ -88,6 +88,21 @@ static reboot_reason_code_t up_reboot_reason_get_hw_value(void)
 		else if ((boot_reason & BIT_BOOT_KM4WDG_RESET_HAPPEN) || (boot_reason & BIT_BOOT_WDG_RESET_HAPPEN)) {
 			return REBOOT_SYSTEM_WATCHDOG;
 		}
+
+		/* KM4 deep sleep handled by KM0 (KM4 sleep + KM0 tickless, KM4 deep sleep + KM0 deep sleep AON) */
+		else if (boot_reason & BIT_BOOT_DSLP_RESET_HAPPEN) {
+			return REBOOT_SYSTEM_DSLP_RESET;
+		}
+
+		/* KM4 or KM0 WDT reset */
```
Comment (jeongchanKim) [inline os/arch/arm/src/amebad/amebad_reboot_reason.c:97]: What is the difference between line 88 and this?
In the comment, they are all from `KM4 or KM0 WDT reset`.

File: os/include/tinyara/reboot_reason.h [hunk 2]
```diff
@@ -40,6 +40,10 @@ typedef enum {
 	REBOOT_SYSTEM_BINARY_UPDATE        = 7,	 /* Reboot for Binary Update */
 	REBOOT_SYSTEM_BINARY_RECOVERYFAIL  = 8,	 /* Binary Recovery Fail */
 
+	REBOOT_SYSTEM_DSLP_RESET           = 27, /* System wake up from deep sleep */
+	REBOOT_SYSTEM_SYS_RESET            = 28, /* System reset by calling sys_reset */
+	REBOOT_SYSTEM_BOD_RESET            = 29, /* Brownout reset */
```
Comment (jeongchanKim) [inline os/include/tinyara/reboot_reason.h:45]: Is there any case for calling `sys_reset` from bootloader?
Within TizenRT side, we all add the reboot reason when calling `sys_reset`.

File: os/arch/arm/src/amebad/amebad_reboot_reason.c (hunk 1 above)
Comment (sangam-swami) [inline os/arch/arm/src/amebad/amebad_reboot_reason.c:97]: This comment from WDT should be sys reset?

File: os/include/tinyara/reboot_reason.h [hunk 3]
```diff
@@ -40,6 +40,10 @@ typedef enum {
 	REBOOT_SYSTEM_BINARY_UPDATE        = 7,	 /* Reboot for Binary Update */
 	REBOOT_SYSTEM_BINARY_RECOVERYFAIL  = 8,	 /* Binary Recovery Fail */
 
+	REBOOT_SYSTEM_DSLP_RESET           = 27, /* System wake up from deep sleep */
+	REBOOT_SYSTEM_SYS_RESET            = 28, /* System reset by calling sys_reset */
```
Comment (jeongchanKim) [inline os/include/tinyara/reboot_reason.h:44]: Is there any case for calling `sys_reset` from bootloader?
Within TizenRT side, we all add the reboot reason when calling `sys_reset`.

Comment (ziyik) [reply os/include/tinyara/reboot_reason.h:44]: Not from bootloader.
But from KM0 (dlp), a prebuild image provided by RTK. (km0_image2_all.bin)

File: os/arch/arm/src/amebad/amebad_reboot_reason.c (hunk 1 above)
Comment (ziyik) [reply os/arch/arm/src/amebad/amebad_reboot_reason.c]: Sorry, is a copy paste mistake.
Corrected.

Comment (ziyik) [reply os/arch/arm/src/amebad/amebad_reboot_reason.c]: Sorry, is a copy paste mistake.
Corrected.

File: os/include/tinyara/reboot_reason.h (hunk 3 above)
Comment (jeongchanKim) [reply os/include/tinyara/reboot_reason.h:44]: Okay. Thanks for checking.
`sys_reset` is only for RTK, not general name, but this is common header.
How about changing the comment like `system reset by core` or something which can explain the case?

Comment (ziyik) [reply os/include/tinyara/reboot_reason.h]: Done change.

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**